README.md

```diff
@@ -310,6 +310,35 @@ The following arguments are passed into your middleware function:
 - suffix: The symbol name suffix (e.g. `/Desktop`)
 - RESIZING_CONSTRAINTS: Object containing constants for the `setResizingConstraint` API
 
+#### Text Middleware
+
+This middleware is executed for every text layer within a document.
+
+```js
+module.exports = {
+  textMiddleware: args => { ... }
+};
+```
+
+The following arguments are passed into your middleware function:
+
+- layer: The current symbol layer
+- node: The source HTML node
+
+Example for preserving text across library updates (by hardcoding an ID):
+
+```js
+textMiddleware: ({ layer, node }) => {
+  const parentNode = node.parentNode;
+  if (parentNode && parentNode.dataset) {
+    const { sketchId } = parentNode.dataset;
+
+    if (sketchId) {
+      layer.setObjectID(`text:${sketchId}`);
```

Comment:
please don't use arbitrary strings as ObjectId, it's going to break Sketch in very subtle ways. Sketch expect IDs to be UUID.

You can generate a fixed UUID from a seed if you want:
```js
import * as seedrandom from 'seedrandom';

const lut = [];
for (let i = 0; i < 256; i += 1) {
  lut[i] = (i < 16 ? '0' : '') + i.toString(16);
}
// Hack (http://stackoverflow.com/a/21963136)
function e7(seed?: ?string) {
  const random = seed ? seedrandom(`${seed}0`) : Math.random;
  const d0 = (random() * 0xffffffff) | 0;
  const d1 = (random() * 0xffffffff) | 0;
  const d2 = (random() * 0xffffffff) | 0;
  const d3 = (random() * 0xffffffff) | 0;
  return `${lut[d0 & 0xff] +
    lut[(d0 >> 8) & 0xff] +
    lut[(d0 >> 16) & 0xff] +
    lut[(d0 >> 24) & 0xff]}-${lut[d1 & 0xff]}${lut[(d1 >> 8) & 0xff]}-${
    lut[((d1 >> 16) & 0x0f) | 0x40]
  }${lut[(d1 >> 24) & 0xff]}-${lut[(d2 & 0x3f) | 0x80]}${lut[(d2 >> 8) & 0xff]}-${
    lut[(d2 >> 16) & 0xff]
  }${lut[(d2 >> 24) & 0xff]}${lut[d3 & 0xff]}${lut[(d3 >> 8) & 0xff]}${lut[(d3 >> 16) & 0xff]}${
    lut[(d3 >> 24) & 0xff]
  }`;
}

// Keep track of previous seeds
const previousSeeds = {};

export function generateID(seed) {
  let _seed;

  if (seed) {
    if (!previousSeeds[seed]) {
      previousSeeds[seed] = 0;
    }
    previousSeeds[seed] += 1;

    _seed = `${seed}${previousSeeds[seed]}`;
  }

  return e7(_seed);
}
```